src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -117,13 +117,13 @@ public function addResource($format, $resource, $locale, $domain = null)
     /**
      * {@inheritdoc}
      */
-    protected function initializeCatalogue($locale)
+    protected function initializeCatalogue(string $locale)
     {
         $this->initialize();
         parent::initializeCatalogue($locale);
     }
 
-    protected function doLoadCatalogue($locale): void
+    protected function doLoadCatalogue(string $locale): void
```

Comment:
shouldn't this be internal (in 4.4 then)
if yes, then let's not add this one and use an annotation instead, to keep compat with translation v4.4

Reply:
That wouldn't change a thing. If we want to maintain compatibility with Translation 4.4, we must not add any of the types I've added to this file. I'd simply leave this PR open until 5.0 has been branched out.

Reply:
ok, then let's still add the missing `@internal` annotation to 4.3